clients/src/main/java/org/apache/kafka/common/security/oauthbearer/OAuthBearerLoginModule.java

```diff
@@ -236,6 +236,13 @@
  * @see SaslConfigs#SASL_LOGIN_REFRESH_BUFFER_SECONDS_DOC
  */
 public class OAuthBearerLoginModule implements LoginModule {
+
+    private enum LoginState {
+        NOT_LOGGED_IN,
```

Comment:
Would be good to comment that this implies logged-out or aborted since the state is null at the beginning (when we aren't logged in, either, but we don't set the state to NOT_LOGGED_IN).  Either comment or rename to LOGGED_OUT_OR_ABORTED?